src/aioquic/h3/connection.py

```diff
@@ -360,6 +362,24 @@ def send_headers(
             stream_id, encode_frame(FrameType.HEADERS, frame_data), end_stream
         )
 
+    def close_connection(self, last_stream_id: int = None) -> None:
+        """
+        Close a connection, emitting a GOAWAY frame.
+        :param last_stream_id: Client-initailized stream ID that can be responsed before connection is closed.
```

Comment:
I'm not sure `close_connection` is very descriptive, since the definition of `GOAWAY` explicitly states: "GOAWAY by itself does not close a connection". The RFC uses the term "shutdown", so maybe "shutdown_connection"?

This docstring needs improvement:

- Mention only servers can call this method
- Typo: "initialized" 
- I'm not sure `last_stream_id` is a good name, as this is in fact the first stream which will never get a response. Maybe simply `stream_id` would be better. Also, why do we want this part of the API? Can't we automatically determine the last stream for which we fully served the response?

Reply:
- 'close_connection' follows H2Connection API, but also I don't like this. I think 'shutdown' is fine. 

- My English is so poor ;-) but I think 'initiated' is more proper than 'initialized' by RFC section-7.2.6)

- 'last_stream_id' is also follow [H2Connection API](https://python-hyper.org/projects/h2/en/stable/api.html#h2.connection.H2Connection.close_connection). And I think it is more explicit and it is good.

   And I can't understand exactly your words. 'last_stream_id' is optional but I think H3Connection should give explicit way for server administrative activity. For example, under exhaust of system memory situation, server need both: 

  - rapid recovering by ignoring some of already received requests
  - but graceful shutdown manner by notifying explicit last stream ID to serve in spite of server emergency

TLTR, 

- I agree to change 'shutdown'
- 'last_stream_id' is better and it is very necessary

Reply:
I had not realized you were mimicking H2Connection's API.. now I have conflicting feelings about naming.

In any case can you please add full test coverage for the new code?

Reply:
I think I already make full test coverage. Anyway, when your master branch became stable, I will merge and retry.